managedtenants/bundles/addon_bundles.py

```diff
@@ -185,17 +185,20 @@ def _get_ocm_config(self):
         """
         ocm_config = self.config.get("ocm", {})
 
-        def ocm_config_getter(k, default):
-            return ocm_config.get(k, default)
+        def ocm_config_getter(k):
+            return ocm_config.get(k)
 
         return {
-            k: ocm_config_getter(k, default)
-            for k, default in [
-                ("addOnParameters", []),
-                ("addOnRequirements", []),
-                ("subOperators", []),
-                ("config", {}),
+            k: ocm_config_getter(k)
```

Comment:
This is nit-picky, but I think you could use an assignment expression here! See point 4 of the first answer https://stackoverflow.com/questions/10291997/how-can-i-do-assignments-in-a-list-comprehension

Reply:
I think it would look like 
`{k: value for k in [...] if (value := ocm_config.get(k))`

Reply:
It was in our style guide to avoid assignment expressions since they force Python `>=3.8` (which doesn't apply anymore), but IMO this would be more readable if it was less compact.

Something like:

```python
...
raw_ocm_config = self.config.get('ocm', {})

_OCM_CONFIG_FIELDS = (
    "addOnParameters",
    "addOnRequirements",
    "subOperators",
    "config",
    "pullSecretName",
    "additionalCatalogSources",
)

return _remove_empty_values(
    _select_fields(_OCM_CONFIG_FIELDS, raw_ocm_config)
)

...

def _select_fields(fields, d):
    return {k: v for k, v in d.items() if k in fields}
    
    
def _remove_empty_values(d):
    return {k: v for k, v in d.items() if v}
```

Reply:
Thanks guys

Reply:
we dont want to remove explicitly set empty values